### ISSUE TITLE
Optimize API objects

### ISSUE THREAD

**Issue by mtbitcr** [BitcreditProtocol/E-Bill#121]

**Comment by zupzup** [comment]
* Introduce API versioning
* Figure out the minimum amount of fields frontend needs for each endpoint
* create `data` module or something similar, which holds the data objects used to communicate with frontend

**Comment by zupzup** [comment]
Meeting Notes 12.11.2024

* for all parameters, we use `snake_case`

# Screens based on Figma

* Create identity
    * Mandatory
        * Legal full name (one field) - (String)
        * email address - (String)
        * postal address - (String)
    * Optional
        * Birth date (Date)
        * Country of birth (String)
        * City of birth (String)
        * Social Security Number (String)
        * Profile Picture
        * Identity document (passport, perso, drivers license) (File)
* Create company
    * Mandatory
        * Legal company name (String)
        * Registration date (of the company) (Date)
        * Country of registration (String)
        * City of registration (String)
        * Postal Address (String)
        * Legal email address (String)
        * Registration Number (String)
        * Proof of registration (File)
    * Optional
        * Logo
* Create contact
    * Mandatory
        * Name (String)
        * Mail (String)
        * Postal Address (String)
        * (`node_id`) (String)
    * Optional
        * Date of birth (Date)
        * Country (String)
        * City (String)
* Issue Bill
    * Mandatory
        * type (Promissory note, three parties, draw on payer) (Enum/String/Code)
        * Country of issuing (String)
        * City of issuing (String)
        * issue date (Date)
        * maturity date (Date)
        * Payee (Contact (node id String))
        * Payer (Contact (node id String))
        * Sum (number)
        * Currency (enum / 3-letter ISO String)
        * Invoice (1-n files)
        * Country of payment (String)
        * City of payment (String)
* Endorse Bill
    * Mandatory
        * Endorsee (Contact (node id String))
* Mint
    * Mandatory
        * Endorsee (Contact (node id String))
        * Sum (number)
        * Currency (enum / "CR-" + 3-letter ISO String)
* Sell
    * Mandatory
        * Endorsee (Contact (node id String))
        * Sum (number)
        * Currency (enum / 3-letter ISO String)
* Bill Overview
    * Input parameters
        * Filter
            * search term (String)
            * date range
                * from (Date)
                * to (Date)
            * Role (Enum - All/Payee/Payer/Contingent)
            * (Currency)
    * Response
        * [Light Bill]
* Bill Detail
    * Full Bill (TBD)
        * latest version of the bill
        * includes all fields of the bill
        * includes calculated current state (e.g. current holder) and possible state transitions (e.g. request to pay/accept etc.)
            * permissions
        * includes full history (maybe on-demand endpoint)
* Home Overview
    * Input parameters
        * Currency
    * Response
        * Balances
            * Payee (number calculated by chosen currency)
            * Payer (number calculated by chosen currency)
            * Contingent (number calculated by chosen currency)
        * N Recent Bills
            * [Light Bill]
                * Payer (Legal full name of person/company)
                * Payee (Legal full name of person/company)
                * Drawer (Legal full name of person/company)
                * Sum (number calculated by chosen currency)
                * Issue Date (Date)


Next step - putting it in Postman

**Comment by mtbitcr** [comment]
> are we staying with form-data for payloads, or will we move to JSON?

I think we must switch to JSON. What's your opinion @zupzup ?

**Comment by zupzup** [comment]
> > are we staying with form-data for payloads, or will we move to JSON?
> 
> I think we must switch to JSON. What's your opinion @zupzup ?

Yes, I think it's more consistent to have JSON everywhere, instead of a mix of JSON and form-data and it's not a big code change. :+1:

**Comment by zupzup** [comment]
API moved to JSON and all current API endpoints documented in Postman.
Waiting for the rest of the screens to continue

**Comment by zupzup** [comment]
Next steps:
* Another iteration round, adding new endpoints according to Figma
* Change postal address to County/City/Address/Zip code
